unknown type name 'cpu_set_t'  on SPARC Solaris 11 and gcc 10

building OpenBlas 0.3.13, the first error is 
```
memory.c: In function 'get_num_procs':
memory.c:225:3: error: unknown type name 'cpu_set_t'
  225 |   cpu_set_t cpuset,*cpusetp;
      |   ^~~~~~~~~
```
indeed, there is no `cpu_set_t`, there is `cpuset_t` only (in `sys/cpuvars.h`)

--- a/driver/others/memory.c
+++ b/driver/others/memory.c
@@ -222,11 +222,11 @@ int get_num_procs(void);
 #else
 int get_num_procs(void) {
   static int nums = 0;
+
+#if defined(__GLIBC_PREREQ)
   cpu_set_t cpuset,*cpusetp;
   size_t size;
   int ret;
-
-#if defined(__GLIBC_PREREQ)
 #if !__GLIBC_PREREQ(2, 7)
   int i;
 #if !__GLIBC_PREREQ(2, 6)

